src/core/abr/network_analyzer.ts

```diff
@@ -103,54 +134,64 @@ function estimateRemainingTime(
  * If that's the case, re-calculate the bandwidth urgently based on
  * this single request.
  * @param {Object} pendingRequests - Current pending requests.
- * @param {Object} clock - Information on the current playback.
+ * @param {Object} playbackInfo - Information on the current playback.
+ * @param {Object|null} currentRepresentation - The Representation being
+ * presently being loaded.
  * @param {Number} lastEstimatedBitrate - Last bitrate estimate emitted.
  * @returns {Number|undefined}
  */
 function estimateStarvationModeBitrate(
   pendingRequests : IRequestInfo[],
-  clock : INetworkAnalizerClockTick,
+  playbackInfo : IPlaybackConditionsInfo,
   currentRepresentation : Representation | null,
   lastEstimatedBitrate : number|undefined
 ) : number|undefined {
-  const nextNeededPosition = clock.currentTime + clock.bufferGap;
-  const concernedRequest = getConcernedRequest(pendingRequests, nextNeededPosition);
-  if (concernedRequest === undefined) {
+  const nextNeededPosition = playbackInfo.currentTime + playbackInfo.bufferGap;
+  const concernedRequests = getConcernedRequests(pendingRequests, nextNeededPosition);
+
+  if (concernedRequests.length !== 1) { // 0  == no request
```

Comment:
Or we could have not returned early and do the following logic when there is only one concerned request. It would maybe avoid to comment for the 0 and > 1 conditions.

Reply:
In that case I think I would still have to explain why I don't want to give an estimate when there's more than one request for that segment. So it doesn't change much the situation here.